lib/SILGen/SILGenPoly.cpp

```diff
@@ -4141,6 +4141,7 @@ void SILGenFunction::emitProtocolWitness(AbstractionPattern reqtOrigTy,
     break;
   }
 
+  formalEvalScope.pop();
```

Comment:
We weren't ever popping the formal eval scope... Was causing my conversion to use formal access cleanups in certain places to trigger the assert about mixing formal evaluation/non-formal evaluation scopes in appropriately.